week-2/README.md

```diff
@@ -0,0 +1,544 @@
+
+```hcl
+variable "name_prefix" {
+  description = "Identifier for the database instance"
+  type        = string
+}
+
+variable "instance_class" {
+  description = "Instance class for the DB instance"
+  type        = string
+  default     = "db.t3.micro"
+}
+
+variable "allocated_storage" {
+  description = "Allocated storage in GB"
+  type        = number
+  default     = 20
+}
+
+variable "engine" {
+  description = "Database engine"
+  type        = string
+  default     = "mariadb"
+}
+
+variable "engine_version" {
+  description = "Database engine version"
+  type        = string
+  default     = "10.6"
+}
+
+variable "db_name" {
+  description = "Name of the database"
+  type        = string
+}
+
+variable "username" {
+  description = "Master username for the database"
+  type        = string
+}
+
+variable "password" {
+  description = "Master password for the database"
+  type        = string
+}
+
+variable "tags" {
+  description = "Tags to apply to the database instance"
+  type        = map(string)
+  default     = {}
+}
+```
+
+##### **`outputs.tf`**
+
+```hcl
+output "endpoint" {
+  description = "Database endpoint"
+  value       = aws_db_instance.this.endpoint
+}
+
+output "username" {
+  description = "Database master username"
+  value       = aws_db_instance.this.username
+}
+
+output "password" {
+  description = "Database master password"
+  value       = aws_db_instance.this.password
+}
+
+output "db_name" {
+  description = "Database name"
+  value       = aws_db_instance.this.db_name
+}
+
+output "security_group_id" {
+  description = "ID of the database security group"
+  value       = aws_security_group.this.id
+}
+```
+
+#### Step 2.4: Return to Root Directory
+
+Navigate back to the root directory:
+
+```sh
+cd ../../
+```
+
+### 3. Using the Modules in Your Configuration
+
+Now that we've created modules for the `aws_instance` and `aws_db_instance`, let's use them in our main configuration.
+
+#### Step 3.1: Set Up the Root Configuration
+
+Create or modify `main.tf` in the root directory:
+
+**`main.tf`**:
+
+```hcl
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.68"
+    }
+  }
+}
+
+provider "aws" {
+  region = "us-west-2"
+
+  default_tags {
+    tags = {
+      environment = "dev"
+      project     = "opentofu-foundations"
+    }
+  }
+}
+
+# Data source to get the default VPC
+data "aws_vpc" "default" {
+  filter {
+    name   = "isDefault"
+    values = ["true"]
+  }
+}
+
+# Module for Database Instance
+module "aws_db_instance" {
+  source = "./modules/aws_db_instance"
+
+  name_prefix = "week2-db"
+  db_name     = "wordpress"
+  username    = "admin"
+  password    = "yourpassword"  # In production, use a secure method for passwords
+
+  ingress_cidr_blocks = [data.aws_vpc.default.cidr_block]
+
+  tags = {
+    Owner = "YourName"
+  }
+}
+
+# Module for EC2 Instance
+module "aws_instance" {
+  source = "./modules/aws_instance"
```

Comment:
I might touch on refs and mention that you can load from other git repos when talking through this part